### DISCUSSION TITLE
Basic FOR loop not working

### DISCUSSION THREAD

**Discussion by son-ra** [gnudatalanguage/gdl#1505] [General]
Hi I am new to IDL/GDL, so apologies if I am missing something but I cannot get FOR loops to work with DO BEGIN.  Here is an example of a script test_GDL.pro. I am using V0.9.9, GNU compilation. 

When I run the following script:
`PRO test_GDL`
`FOR i=0,5 DO PRINT, i`
`end`

I get some messages, but ultimately the output I asked for:
`% Programs can't be compiled from single statement mode.`
`       0`
`       1`
`      2`
`      3`
`       4`
`       5`
`% Parser syntax error: unexpected token: END`

However, when I run this script:
`PRO test_GDL`
`FOR i=0,5 DO BEGIN`
`PRINT, i`
`ENDFOR`
`end`

I get this issue:
`% Programs can't be compiled from single statement mode.`
`% Parser syntax error: unexpected end of file`
`       6`
`% Parser syntax error: unexpected token: ENDFOR`
`% Parser syntax error: unexpected token: END`

Could anyone provide insight into what is going on here? Thank you!

**Comment by slayoo** [comment]
This discussion seems to be relevant: https://github.com/gnudatalanguage/gdl/discussions/1504 (based on https://stackoverflow.com/questions/64335330/error-when-defining-function-in-gdl-on-ubuntu-20)
HTH

**Reply by son-ra** [reply]
Thank you so much!! It was indeed because I was running my scripts using @[script_name].pro instead of [script_name]

**Comment by GillesDuvert** [comment]
Better, see [here](https://stackoverflow.com/questions/38364599/syntax-error-on-foreach-endforeach-loop)
In particular:
`Batch files (called via @myscript) can't have compound statements, i.e. with BEGIN/END. Make it into a procedure/function or a main-level program.`
